Auto-Wrapping Disposals Chutes + Map Tweaks

CPU-Blanc opened the pull request:
This PR aims to make the process of sending ammo from cargo to the various hardpoints much easier as well as a few tweaks

Changes:
- Added an auto-wrapping chute to replace the regular chute in the lower cargo bay. This is to match the unwrappers introduced in the various gun bays. These will wrap items on bump, and tag them with the destination set on the chute. Clicking the chute with an empty hand brings up the same menu you would see with the handheld taggers (I'm lazy and didn't want to make a whole new UI). Now random torpedoes won't end up in the mailing office/disposals! This should also make making batches of ammo much easier.
- Reverted the merging of the torpedo and gun bay disposal lines from the main loop in order to facilitate faster movement of ammo from cargo to their intended weapons. This will need to be double-checked as the disposals loop is a bit of a maze, but I've done my best to revert it to how it used to be, while adding a new dedicated line for the hardpoints. If this is unwanted, let me know and I'll merge them back again.
- Separated the cargo conveyor belt switch from the torpedo loading belt.
- Changed the switches in both gun bays to one directional  to bring them in line with other loading belts.
- Removed duplicate lattice outside the Starboard Gun Bay, which has been causing a runtime on every boot.
- Added `throw_at()` proc call to atoms being vented from unwrapper outlets to bring them in line with other outlets. This has been dialled back to a throw distance of 1 however to avoid issues in the gun bays. This also allows torpedoes to be placed directly on the belt, as previously it would exit onto the floor and would need to be dragged to load or get onto the belt.

Any issues, let me know!

Glloyd commented:
I had merged the torpedo line with the main line to make shipping to the new gun bays less of a mess disposal-wise, and also to facilitate shipping things directly from the workshop into the main loop. This way also allows other parts of the ship to send things to the gun bays if need be (I was thinking in particular of sending weapons there through the main loop when I made that change) It's set up so the trip to the torpedo bay should be quite short from the workshop, although the trip to the gun bays will take a bit longer, so something could be done there to speed things up while still keeping it all in the main loop.

CPU-Blanc commented:
As it currently stands, the weapons loop does actually connect back to the main loop under the meeting room although it's only 1-way (workshop can go to main loop after the weapons loop, main loop can't go to weapons). I can always put disposals back if you'd rather it stay how it is though!

Glloyd commented:
Yeah, I'd prefer to keep it so the main loop can go to the weapons, so that
things can be sent there from elsewhere, and so that drones can travel
there directly. This came up a while back and was part of the reason I
merged it.

On Fri, Mar 11, 2022 at 10:54 AM CPU_Blanc ***@***.***> wrote:

> As it currently stands, the weapons loop does actually connect back to the
> main loop under the meeting room although it's only 1-way (workshop can go
> to main loop after the weapons loop, main loop can't go to weapons). I can
> always put disposals back if you'd rather it stay how it is though!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/UristMcStation/UristMcStation/pull/1052#issuecomment-1065241607>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA6K6WA6LB2COOLS6VV3WATU7NUCTANCNFSM5PJDJKMQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Graham Lloyd

CPU-Blanc commented:
No worries! Reverted the disposal loop changes. While doing so, I discovered and fixed another bug.

`/obj/structure/disposalholder/` would continue to call `transfer()` via it's `Process()` during the sleep period for the expel animation. This would result in the `expel()` proc being called multiple times with every tick; stacking all sounds and making the flick animation be incorrect. Corrected it by removing the holder's process

Glloyd commented:
Looks good!
